hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java

```diff
@@ -276,6 +283,153 @@ public void remoteMap() {
+        BatchSource<Integer> source1 = TestSources.items(input);
+        String dataConnectionName = randomString();
+        // When
+        Sink<Entry<String, Integer>> sink = sinkFn.apply(dataConnectionName);
+        p.readFrom(source1)
+                .map(i -> entry(String.valueOf(i), i))
+                .writeTo(sink);
+        // Then
+        Job job = hz().getJet().newJob(p, new JobConfig());
+        assertThatThrownBy(job::join).isInstanceOf(CompletionException.class).hasStackTraceContaining("Data connection '" + dataConnectionName + "' not found");
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionToNonExistentCluster() {
+        remoteReplicatedMap_dataConnectionToNonExistentCluster((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionToNonExistentCluster_customBatchSize() {
+        remoteReplicatedMap_dataConnectionToNonExistentCluster((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName, 100));
+    }
+
+    private void remoteReplicatedMap_dataConnectionToNonExistentCluster(Function<String, Sink<Entry<String, Integer>>> sinkFn) {
+        // Given
+        List<Integer> input = sequence(itemCount);
+        BatchSource<Integer> source1 = TestSources.items(input);
+        // When
+        String dataConnectionName = randomString();
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.setClusterName("neverland");
+        clientConfig.getNetworkConfig().addAddress("localhost:911");
+        clientConfig.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(3000);
+        hz().getConfig().addDataConnectionConfig(new DataConnectionConfig(dataConnectionName)
+                .setType("Hz")
+                .setShared(false)
+                .setProperty(HazelcastDataConnection.CLIENT_XML, ImdgUtil.asXmlString(clientConfig)));
+        Sink<Entry<String, Integer>> sink = sinkFn.apply(dataConnectionName);
+        p.readFrom(source1)
+                .map(i -> entry(String.valueOf(i), i))
+                .writeTo(sink);
+        Job job = hz().getJet().newJob(p, new JobConfig());
+        assertThatThrownBy(job::join).isInstanceOf(CompletionException.class).hasStackTraceContaining("Unable to connect");
+    }
+
+    @Test
+    public void remoteReplicatedMap_noItemsInSource() {
+        remoteReplicatedMap_noItemsInSource((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_noItemsInSource_customBatchSize() {
+        remoteReplicatedMap_noItemsInSource((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName, 100));
+    }
+
+    private void remoteReplicatedMap_noItemsInSource(Function<String, Sink<Entry<String, Integer>>> sinkFn) {
+        // Given
+        BatchSource<Integer> source1 = TestSources.items();
```

Comment:
nit:
why `source1` not just `source`? Maybe some leftover (but it's in all tests).

Reply:
choice of github copilot :) but let me change it.